static/app/views/settings/components/forms/choiceMapperField.tsx

```diff
@@ -140,8 +140,9 @@ export default class ChoiceMapper extends React.Component<FieldProps> {
 
     const removeRow = (itemKey: string) => {
       // eslint-disable-next-line no-unused-vars
-      const {[itemKey]: _, ...updatedValue} = value;
-      saveChanges(updatedValue);
+      saveChanges(
+        Object.fromEntries(Object.entries(value).filter(([key, _]) => key !== itemKey))
+      );
```

Comment:
I think we should put the _saveChanges_ param to a clearer identifier or write a comment explaining what is happening here for readability purpose.